app/assets/javascripts/smoothState.js

```diff
@@ -0,0 +1,746 @@
+  }
+
+  /** Abort if smoothState is already present **/
+  if($.fn.smoothState) { return; }
+
+  var
+    /** Used later to scroll page to the top */
+    $body = $('html, body'),
+
+    /** Used in debug mode to console out useful warnings */
+    consl = window.console,
+
+    /** Plugin default options, will be exposed as $fn.smoothState.options */
+    defaults = {
+
+      /** If set to true, smoothState will log useful debug information instead of aborting */
+      debug: false,
+
+      /** jQuery selector to specify which anchors smoothState should bind to */
+      anchors: 'a',
+
+	  /** Regex to specify which anchor's with a specific href value based on the regex smoothState should bind to. If empty, every href will be permitted. */
+	  hrefRegex: '',
+
+      /** jQuery selector to specify which forms smoothState should bind to */
+      forms: 'form',
+
+      /** If set to true, smoothState will store form responses in the cache. */
+      allowFormCaching: false,
+
+      /** Minimum number of milliseconds between click/submit events. Events ignored beyond this rate are ignored. */
+      repeatDelay: 500,
+
+      /** A selector that defines what should be ignored by smoothState */
+      blacklist: '.no-smoothState',
+
+      /** If set to true, smoothState will prefetch a link's contents on hover */
+      prefetch: false,
+
+      /** The name of the event we will listen to from anchors if we're prefetching */
+      prefetchOn: 'mouseover touchstart',
+
+      /** The number of pages smoothState will try to store in memory */
+      cacheLength: 0,
+
+      /** Class that will be applied to the body while the page is loading */
+      loadingClass: 'is-loading',
+
+      /** Scroll to top after onStart and scroll to hash after onReady */
+      scroll: true,
+
+      /**
+       * A function that can be used to alter the ajax request settings before it is called
+       * @param  {Object} request jQuery.ajax settings object that will be used to make the request
+       * @return {Object}         Altered request object
+       */
+      alterRequest: function (request) {
+        return request;
+      },
+
+      /** Run before a page load has been activated */
+      onBefore: function ($currentTarget, $container) {},
+
+      /** Run when a page load has been activated */
+      onStart: {
+        duration: 0,
+        render: function ($container) {}
+      },
+
+      /** Run if the page request is still pending and onStart has finished animating */
+      onProgress: {
+        duration: 0,
+        render: function ($container) {}
+      },
+
+      /** Run when requested content is ready to be injected into the page  */
+      onReady: {
+        duration: 0,
+        render: function ($container, $newContent) {
+          $container.html($newContent);
+        }
+      },
+
+      /** Run when content has been injected and all animations are complete  */
+      onAfter: function($container, $newContent) {}
```

Comment:
'$newContent' is defined but never used.<br>'$container' is defined but never used.